src/script/sigencoding.cpp

```diff
@@ -215,7 +215,8 @@ static bool CheckSighashEncoding(const valtype &vchSig, uint32_t flags,
         return set_error(serror, ScriptError::SIG_HASHTYPE);
     }
 
-    bool usesForkId = GetHashType(vchSig).hasForkId();
+    SigHashType sigHashType = GetHashType(vchSig);
+    bool usesForkId = sigHashType.hasForkId() || sigHashType.hasBIP341();
```

Comment:
!legacy?

Reply:
There's also `isReserved`; I'm being a bit conservative here, otherwise we get SIGHASH_BUG.